src/app/reporting/ReportingEngine.h

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines Reporting Engine for a CHIP Interaction Model
+ *
+ */
+
+#pragma once
+
+#ifndef _CHIP_DATA_MODEL_REPORTING_ENGINE_H
+#define _CHIP_DATA_MODEL_REPORTING_ENGINE_H
+
+#include <app/MessageDef/ReportData.h>
+#include <app/ReadHandler.h>
+#include <app/reporting/EventLoggingTypes.h>
+#include <app/reporting/LoggingManagement.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <map>
```

Comment:
is map safe to use in embedded?

Reply:
not yet used map here, just removed, thanks

Reply:
I may be missing an update - this still exists in the latest upload.

Reply:
Done, thanks